### DISCUSSION TITLE
API reporting ip address as {"public_ip":""}

### DISCUSSION THREAD

**Discussion by GabrielPearce** [qdm12/gluetun#2088] [Q&A]
I use Homepage and noticed this yesterday. 
![image](https://github.com/qdm12/gluetun/assets/46075675/9536f258-bb49-4602-9c7b-a9e65ede3da5)
the container was working fine yesterday but this has started to occur. 

I looked at the logs and noticed

2024-02-01T12:53:50Z WARN [ip getter] too many requests sent for this month from https://ipinfo.io/: 429 429 Too Many Requests; not retrying.

A solution would be to allow us to config our own external ip identifier. I know ifconfig.co and icanhazip.com are both quite usable.
(curl ifconfig.co should return your public ip address)

perhaps it could be an enviroment vairable we can set?

**Comment by kserge1t** [comment]
Same here (also with [Homepage Gluetun Widget](https://gethomepage.dev/widgets/services/gluetun/)), was working fine yesterday, and now returns {"public_ip":""}

![image](https://github.com/user-attachments/assets/14fe401a-36af-4944-98fe-f69354c1aaab)
